diff --git a/headless/BUILD.gn b/headless/BUILD.gn
index 7aec54864b..9756330638 100644
--- a/headless/BUILD.gn
+++ b/headless/BUILD.gn
@@ -341,6 +341,7 @@ component("headless_non_renderer") {
 
   deps = [
     ":backend_cdp_bindings",
+    "//abrowser/src/browser:abrowser",
     "//base:base_static",
     "//build:branding_buildflags",
     "//build:chromeos_buildflags",
@@ -922,7 +923,10 @@ executable("headless_shell") {
   sources = [ "app/headless_shell_main.cc" ]
   defines = []
 
-  deps = [ ":headless_shell_lib" ]
+  deps = [
+    ":headless_shell_lib",
+    "//abrowser/src/browser:abrowser",
+  ]
 
   if (proprietary_codecs) {
     deps += [ ":generate_headless_shell_license_file" ]
