diff --git a/chrome/BUILD.gn b/chrome/BUILD.gn
index e08a382947..552f5a8ca0 100644
--- a/chrome/BUILD.gn
+++ b/chrome/BUILD.gn
@@ -982,13 +982,15 @@ if (is_win) {
     }
 
     # Add the SwiftShader .dylibs in the MODULE_DIR of Chromium.app
-    bundle_data("swiftshader_binaries") {
-      sources = [
-        "$root_out_dir/vk_intermediates/libvk_swiftshader.dylib",
-        "$root_out_dir/vk_intermediates/vk_swiftshader_icd.json",
-      ]
-      outputs = [ "{{bundle_contents_dir}}/Libraries/{{source_file_part}}" ]
-      public_deps = [ "//ui/gl:swiftshader_vk_library_copy" ]
+    if (enable_swiftshader) {
+      bundle_data("swiftshader_binaries") {
+        sources = [
+          "$root_out_dir/vk_intermediates/libvk_swiftshader.dylib",
+          "$root_out_dir/vk_intermediates/vk_swiftshader_icd.json",
+        ]
+        outputs = [ "{{bundle_contents_dir}}/Libraries/{{source_file_part}}" ]
+        public_deps = [ "//ui/gl:swiftshader_vk_library_copy" ]
+      }
     }
   }
 
@@ -999,7 +1001,7 @@ if (is_win) {
   }
 
   group("swiftshader_library") {
-    if (bundle_egl_libraries) {
+    if (bundle_egl_libraries && enable_swiftshader) {
       deps = [ ":swiftshader_binaries" ]
     }
   }
diff --git a/content/shell/BUILD.gn b/content/shell/BUILD.gn
index 3b7d6008ac..3a3f9878c1 100644
--- a/content/shell/BUILD.gn
+++ b/content/shell/BUILD.gn
@@ -838,13 +838,15 @@ if (is_mac) {
     }
 
     # Add the SwiftShader .dylibs in the Libraries directory of the Framework.
-    bundle_data("content_shell_swiftshader_binaries") {
-      sources = [
-        "$root_out_dir/vk_intermediates/libvk_swiftshader.dylib",
-        "$root_out_dir/vk_intermediates/vk_swiftshader_icd.json",
-      ]
-      outputs = [ "{{bundle_contents_dir}}/Libraries/{{source_file_part}}" ]
-      public_deps = [ "//ui/gl:swiftshader_vk_library_copy" ]
+    if (enable_swiftshader) {
+      bundle_data("content_shell_swiftshader_binaries") {
+        sources = [
+          "$root_out_dir/vk_intermediates/libvk_swiftshader.dylib",
+          "$root_out_dir/vk_intermediates/vk_swiftshader_icd.json",
+        ]
+        outputs = [ "{{bundle_contents_dir}}/Libraries/{{source_file_part}}" ]
+        public_deps = [ "//ui/gl:swiftshader_vk_library_copy" ]
+      }
     }
   }
 
@@ -855,7 +857,7 @@ if (is_mac) {
   }
 
   group("content_shell_swiftshader_library") {
-    if (use_egl) {
+    if (use_egl && enable_swiftshader) {
       deps = [ ":content_shell_swiftshader_binaries" ]
     }
   }
