diff --git a/gpu/command_buffer/service/shared_image/iosurface_image_backing.h b/gpu/command_buffer/service/shared_image/iosurface_image_backing.h
index aee115b016..33a3538d4f 100644
--- a/gpu/command_buffer/service/shared_image/iosurface_image_backing.h
+++ b/gpu/command_buffer/service/shared_image/iosurface_image_backing.h
@@ -9,7 +9,10 @@
 #include "base/containers/flat_map.h"
 #include "base/memory/raw_ptr.h"
 #include "build/build_config.h"
+#include "skia/buildflags.h"
+#if BUILDFLAG(SKIA_USE_DAWN)
 #include "gpu/command_buffer/service/shared_image/dawn_shared_texture_holder.h"
+#endif
 #include "gpu/command_buffer/service/shared_image/shared_image_backing.h"
 #include "gpu/command_buffer/service/texture_manager.h"
 #include "gpu/config/gpu_preferences.h"
@@ -129,8 +132,10 @@ class GPU_GLES2_EXPORT IOSurfaceImageBacking
 
   bool InitializePixels(base::span<const uint8_t> pixel_data);
 
+#if BUILDFLAG(SKIA_USE_DAWN)
   void AddWGPUDeviceWithPendingCommands(wgpu::Device device);
   void WaitForDawnCommandsToBeScheduled(const wgpu::Device& device_to_exclude);
+#endif
 
   void AddEGLDisplayWithPendingCommands(gl::GLDisplayEGL* display);
   void WaitForANGLECommandsToBeScheduled();
@@ -141,7 +146,9 @@ class GPU_GLES2_EXPORT IOSurfaceImageBacking
 
  private:
   class GLTextureIRepresentation;
+#if BUILDFLAG(SKIA_USE_DAWN)
   class DawnRepresentation;
+#endif
   class SkiaGaneshRepresentation;
   class SkiaGraphiteRepresentation;
   class OverlayRepresentation;
@@ -164,6 +171,7 @@ class GPU_GLES2_EXPORT IOSurfaceImageBacking
   std::unique_ptr<OverlayImageRepresentation> ProduceOverlay(
       SharedImageManager* manager,
       MemoryTypeTracker* tracker) final;
+#if BUILDFLAG(SKIA_USE_DAWN)
   std::unique_ptr<DawnImageRepresentation> ProduceDawn(
       SharedImageManager* manager,
       MemoryTypeTracker* tracker,
@@ -171,6 +179,7 @@ class GPU_GLES2_EXPORT IOSurfaceImageBacking
       wgpu::BackendType backend_type,
       std::vector<wgpu::TextureFormat> view_formats,
       scoped_refptr<SharedContextState> context_state) final;
+#endif
   std::unique_ptr<SkiaGaneshImageRepresentation> ProduceSkiaGanesh(
       SharedImageManager* manager,
       MemoryTypeTracker* tracker,
@@ -211,12 +220,15 @@ class GPU_GLES2_EXPORT IOSurfaceImageBacking
   const uint32_t io_surface_format_;
   const gfx::GenericSharedMemoryId io_surface_id_;
 
+#if BUILDFLAG(SKIA_USE_DAWN)
   // DawnSharedTextureHolder that keeps an internal cache of per-device
   // SharedTextureData that vends WebGPU textures for the underlying IOSurface.
   std::unique_ptr<DawnSharedTextureHolder> dawn_texture_holder_;
 
   DawnSharedTextureHolder* GetDawnTextureHolder();
+#endif
 
+#if BUILDFLAG(SKIA_USE_DAWN)
   // Tracks the number of currently-ongoing accesses to a given WGPU texture.
   base::flat_map<WGPUTexture, int> wgpu_texture_ongoing_accesses_;
 
@@ -237,6 +249,7 @@ class GPU_GLES2_EXPORT IOSurfaceImageBacking
   // Returns the number of ongoing accesses that will still be present on this
   // texture after ending this access.
   int TrackEndAccessToWGPUTexture(wgpu::Texture texture);
+#endif
 
   const GLenum gl_target_;
   const bool framebuffer_attachment_angle_;
diff --git a/gpu/command_buffer/service/shared_image/iosurface_image_backing.mm b/gpu/command_buffer/service/shared_image/iosurface_image_backing.mm
index e6e9f7b312..638858120d 100644
--- a/gpu/command_buffer/service/shared_image/iosurface_image_backing.mm
+++ b/gpu/command_buffer/service/shared_image/iosurface_image_backing.mm
@@ -11,7 +11,6 @@
 
 #include <EGL/egl.h>
 #import <Metal/Metal.h>
-#include <dawn/native/MetalBackend.h>
 
 #include "base/apple/scoped_cftyperef.h"
 #include "base/apple/scoped_nsobject.h"
@@ -22,14 +21,18 @@
 #include "components/viz/common/resources/shared_image_format_utils.h"
 #include "gpu/command_buffer/common/shared_image_trace_utils.h"
 #include "gpu/command_buffer/common/shared_image_usage.h"
-#include "gpu/command_buffer/service/dawn_context_provider.h"
 #include "gpu/command_buffer/service/shared_context_state.h"
 #include "gpu/command_buffer/service/shared_image/copy_image_plane.h"
-#include "gpu/command_buffer/service/shared_image/dawn_fallback_image_representation.h"
 #include "gpu/command_buffer/service/shared_image/iosurface_image_backing_factory.h"
 #include "gpu/command_buffer/service/shared_image/shared_image_format_service_utils.h"
 #include "gpu/command_buffer/service/shared_image/shared_image_gl_utils.h"
+#include "skia/buildflags.h"
+#if BUILDFLAG(SKIA_USE_DAWN)
+#include <dawn/native/MetalBackend.h>
+#include "gpu/command_buffer/service/dawn_context_provider.h"
+#include "gpu/command_buffer/service/shared_image/dawn_fallback_image_representation.h"
 #include "gpu/command_buffer/service/shared_image/skia_graphite_dawn_image_representation.h"
+#endif
 #include "gpu/command_buffer/service/skia_utils.h"
 #include "gpu/config/gpu_finch_features.h"
 #include "third_party/skia/include/core/SkColorSpace.h"
@@ -657,9 +660,11 @@ void SetBeginReadAccessCallback(
   // WaitForANGLECommandsToBeScheduled() call is required.
   iosurface_backing->WaitForANGLECommandsToBeScheduled();
 
+#if BUILDFLAG(SKIA_USE_DAWN)
   // Likewise do the same for Dawn's commands.
   iosurface_backing->WaitForDawnCommandsToBeScheduled(
       /*device_to_exclude=*/nullptr);
+#endif
 
   gl::GLContext* context = gl::GLContext::GetCurrent();
   if (context) {
@@ -702,6 +707,7 @@ void SetBeginReadAccessCallback(
   return IOSurfaceIsInUse(io_surface_.get());
 }
 
+#if BUILDFLAG(SKIA_USE_DAWN)
 ///////////////////////////////////////////////////////////////////////////////
 // DawnRepresentation
 
@@ -952,6 +958,7 @@ void SetBeginReadAccessCallback(
     SupportsMultipleConcurrentReadAccess() {
   return true;
 }
+#endif  // BUILDFLAG(SKIA_USE_DAWN)
 
 ///////////////////////////////////////////////////////////////////////////////
 // IOSurfaceImageBacking
@@ -988,12 +995,14 @@ void SetBeginReadAccessCallback(
                        IOSurfaceGetHeight(io_surface_.get())),
       io_surface_format_(IOSurfaceGetPixelFormat(io_surface_.get())),
       io_surface_id_(io_surface_id),
-      dawn_texture_holder_(std::make_unique<DawnSharedTextureHolder>()),
       gl_target_(gl_target),
       framebuffer_attachment_angle_(framebuffer_attachment_angle),
       cleared_rect_(is_cleared ? gfx::Rect(size) : gfx::Rect()),
       gr_context_type_(gr_context_type),
       weak_factory_(this) {
+#if BUILDFLAG(SKIA_USE_DAWN)
+  dawn_texture_holder_ = std::make_unique<DawnSharedTextureHolder>();
+#endif
   CHECK(io_surface_);
 
   // If this will be bound to different GL backends, then make RetainGLTexture
@@ -1029,7 +1038,9 @@ void SetBeginReadAccessCallback(
 
   // Make sure any pending ANGLE EGLDisplays and Dawn devices are flushed.
   WaitForANGLECommandsToBeScheduled();
+#if BUILDFLAG(SKIA_USE_DAWN)
   WaitForDawnCommandsToBeScheduled(/*device_to_exclude=*/nullptr);
+#endif
 
   ScopedIOSurfaceLock io_surface_lock(io_surface_.get(), /*options=*/0);
 
@@ -1080,7 +1091,9 @@ void SetBeginReadAccessCallback(
 
   // Make sure any pending ANGLE EGLDisplays and Dawn devices are flushed.
   WaitForANGLECommandsToBeScheduled();
+#if BUILDFLAG(SKIA_USE_DAWN)
   WaitForDawnCommandsToBeScheduled(/*device_to_exclude=*/nullptr);
+#endif
 
   ScopedIOSurfaceLock io_surface_lock(io_surface_.get(), /*options=*/0);
 
@@ -1258,6 +1271,7 @@ new IOSurfaceBackingEGLState(this, egl_display, context,
                                                  io_surface_);
 }
 
+#if BUILDFLAG(SKIA_USE_DAWN)
 int IOSurfaceImageBacking::TrackBeginAccessToWGPUTexture(
     wgpu::Texture texture) {
   return wgpu_texture_ongoing_accesses_[texture.Get()]++;
@@ -1305,6 +1319,7 @@ new IOSurfaceBackingEGLState(this, egl_display, context,
     wgpu_devices_pending_flush_.insert(device_to_exclude);
   }
 }
+#endif
 
 void IOSurfaceImageBacking::AddEGLDisplayWithPendingCommands(
     gl::GLDisplayEGL* display) {
@@ -1326,6 +1341,7 @@ new IOSurfaceBackingEGLState(this, egl_display, context,
   }
 }
 
+#if BUILDFLAG(SKIA_USE_DAWN)
 std::unique_ptr<DawnImageRepresentation> IOSurfaceImageBacking::ProduceDawn(
     SharedImageManager* manager,
     MemoryTypeTracker* tracker,
@@ -1394,6 +1410,7 @@ new IOSurfaceBackingEGLState(this, egl_display, context,
       manager, this, tracker, wgpu::Device(device), wgpu_format,
       std::move(view_formats));
 }
+#endif  // BUILDFLAG(SKIA_USE_DAWN)
 
 std::unique_ptr<SkiaGaneshImageRepresentation>
 IOSurfaceImageBacking::ProduceSkiaGanesh(
@@ -1590,7 +1607,9 @@ new IOSurfaceBackingEGLState(this, egl_display, context,
   // IOSurface might be written on a different GPU. So we have to wait for the
   // previous Dawn and ANGLE commands to be scheduled first.
   // TODO(crbug.com/40260114): Skip this if we're not on a dual-GPU system.
+#if BUILDFLAG(SKIA_USE_DAWN)
   WaitForDawnCommandsToBeScheduled(/*device_to_exclude=*/nullptr);
+#endif
 
   // Note that we don't need to call WaitForANGLECommandsToBeScheduled for other
   // EGLDisplays because it is already done when the previous GL context is made
